sentiment_analysis.ipynb

```diff
@@ -0,0 +1,219 @@
+    "# TODO: Explain LayerConfig and bolt.Network in markdown.\n",
+    "from thirdai import bolt\n",
+    "\n",
+    "layers = [\n",
+    "    \n",
+    "    bolt.LayerConfig(\n",
+    "        dim=2000, \n",
+    "        load_factor=0.2, \n",
+    "        activation_function=bolt.ActivationFunctions.ReLU),\n",
+    "        \n",
+    "    bolt.LayerConfig(\n",
+    "        dim=2,\n",
+    "        load_factor=1.0, \n",
+    "        activation_function=bolt.ActivationFunctions.Softmax)     \n",
+    "]\n",
+    "\n",
+    "network = bolt.Network(\n",
+    "    layers=layers, \n",
+    "    input_dim=100000)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 3. Training\n",
+    "**The train() method**\n",
+    "\n",
+    "Train the BOLT network by calling the `train()` method, which accepts the following arguments:\n",
+    "* `train_data`: BOLT dataset - The training dataset in a format returned by `dataset.load_bolt_svm_dataset()`.\n",
+    "* `loss_fn`: BOLT loss function - The loss function to minimize. In this demo version, we only support the `bolt.CategoricalCrossEntropyLoss()` loss function.\n",
+    "* `learning_rate`: Float - The learning rate for gradient descent. The default value is 0.0001.\n",
+    "* `epochs`: Int - The number of training epochs (a full cycle through the dataset).\n",
+    "* `verbose` (Optional): Boolean - Set to `True` to print a progress bar, accuracy, and elapsed time for each training epoch. Set to `False` otherwise. `True` by default.\n",
+    "\n",
+    "It then returns a dictionary that contains the loss value and elapsed time for each training epoch.\n",
+    "\n",
+    "\n",
+    "**Training with sparse inference in mind**\n",
+    "\n",
+    "Call the `enable_sparse_inference()` method if you plan to use sparse inference. We recommend training the network for at least one more epoch after calling this method.\n",
+    "\n",
+    "\n",
+    "**Saving a trained model**\n",
+    "\n",
+    "Simply call the `save()` method, passing in the location of the save file."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# TODO: Explain enable sparse inference call.\n",
+    "network.train(\n",
+    "    train_data=train_data,\n",
+    "    loss_fn=bolt.CategoricalCrossEntropyLoss(), \n",
+    "    learning_rate=0.0001, \n",
+    "    epochs=20, \n",
+    "    verbose=True)\n",
+    "\n",
+    "network.enable_sparse_inference()\n",
+    "\n",
+    "network.train(\n",
+    "    train_data=train_data,\n",
+    "    loss_fn=bolt.CategoricalCrossEntropyLoss(), \n",
+    "    learning_rate=0.0001, \n",
+    "    epochs=1,\n",
+    "    verbose=True)\n",
+    "\n",
+    "network.save(filename=\"path/to/savefile\") # Don't forget to change the filename parameter!"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### **4. Inference**\n",
```

Comment:
Where do we actually do the prediction? I don't see it here (I'm probably just missing it in the format). Also, can we do an interactive demo like in my notebook?